### ISSUE TITLE
msi deployment via GPO

### ISSUE THREAD

**Issue by ernestbevin** [sqlitebrowser/sqlitebrowser#2159]
What is your question
-----------------------

Good afternoon:

I have an enquire that when deploying the MSI via GPO, the app installed but does not show the shortcut on the Desktop and program menu for DB Browser and DB Browser (SQLCipher).
also 
I would like to know what is the command for the shortcut or possible either from the MDT deployment
msiexec /i DB.Browser.for.SQLite-3.11.2-win64 /

many thanks

**Comment by justinclift** [comment]
Not sure what to make of this.  @karim or @glawrence Any ideas? :smile:

**Comment by ambersnow** [comment]
Ditto. I am encountering the same issue when deploying the MSI silently, the application installs fine but it does not make any file extension registrations or shortcuts so you cannot access the program without manually digging through Program Files and making a shortcut to the executable directly. 

This is using the latest version with the `msiexec /i DBBrowser.msi /quiet`, as shown in your docs

**Comment by karim** [comment]
Passing options to the MSI installer can be done using the `PROPERTY=value`.

The following are supported:
- `SHORTCUT_SQLITE_DESKTOP`
- `SHORTCUT_SQLCIPHER_DESKTOP`
- `SHORTCUT_SQLITE_PROGRAMMENU`
- `SHORTCUT_SQLCIPHER_PROGRAMMENU`

To enable desktop shortcuts for both SQLite and SQLCipher you can do this:

`msiexec /i DB.Browser.for.SQLite-2020-01-17-win64.msi SHORTCUT_SQLITE_DESKTOP=1 SHORTCUT_SQLCIPHER_DESKTOP=1`.

I'm sorry if this wasn't mentioned anywhere. :smiley:

**Comment by ambersnow** [comment]
Yes! Thank you, this is helpful.
For the people in the future who may encounter it, is it possible to add it to your wiki where the silent install section is: https://github.com/sqlitebrowser/sqlitebrowser/wiki/Silent-Mode-installation-on-Windows

**Comment by ernestbevin** [comment]
that is working on manual MSI install, thank you 
msiexec /i DB4S.msi SHORTCUT_SQLITE_DESKTOP=1 SHORTCUT_SQLCIPHER_DESKTOP=1

for passive install i have use the following
msiexec /i DB4S.msi SHORTCUT_SQLITE_DESKTOP=1 SHORTCUT_SQLCIPHER_DESKTOP=1 /passive

I will see if I can apply the command on GPO & also the MDT. will keep in post.

**Comment by ernestbevin** [comment]
btw karim,
how did you found the command line for the shortcut to the desktop?

**Comment by ernestbevin** [comment]
GPO-installation, only able to push the installation onto the client (as GPO does not have the options for parameters) hence there are no shortcut for 
SHORTCUT_SQLITE_DESKTOP
SHORTCUT_SQLCIPHER_DESKTOP
SHORTCUT_SQLITE_PROGRAMMENU
SHORTCUT_SQLCIPHER_PROGRAMMENU

**Comment by justinclift** [comment]
> For the people in the future who may encounter it, is it possible to add it to your wiki where the silent install section is ...

Good idea, just cut-n-pasted that info from @karim to the wiki page. :smile:

**Comment by justinclift** [comment]
@ernestbevin Does this help at all?

&nbsp; &nbsp; https://social.technet.microsoft.com/Forums/sharepoint/en-US/b3e7bacd-1c33-4d72-a9c6-5a641968c7b1/gpo-to-deploy-an-msi-package-that-also-needs-command-line-options

Note, I've no clue with GPO (etc). :angel:

**Comment by karim** [comment]
> how did you found the command line for the shortcut to the desktop?

`msiexec /?` will show you all the command-line options for the MSI installer. `PROPERTY=value` will set all public properties. Public properties **must** be in CAPS so they can be set on the command-line.

If you mean [SHORTCUT_SQLITE_DESKTOP](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/installer/windows/product.wxs#L113) then I know it because I wrote it. :smiley: I made shortcuts configurable (using conditions) so I gave them names. You can also change the installation directory using `INSTALLDIR=C:\path`.

Sorry, I'm not familiar with GPO so I cannot help.

Thanks Justin for updating the wiki. :smile:

**Comment by ernestbevin** [comment]
thank you Karim:
I thought this would be a standard MSI parameters of SHORTCUT_SQLITE_DESKTOP=1 SHORTCUT_SQLCIPHER_DESKTOP
Would this be next update when deploy MSI can push the shortcut to the desktop and menu
many thanks 
In meanwhile I will see if I can create a shortcut via GPO .

Thank you for everyone for the quick response.

**Comment by karim** [comment]
> Would this be next update

Yes, and it should be very soon. :tada: 

> In meanwhile I will see if I can create a shortcut via GPO .

Please report your findings here so we can close this issue.
